app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -23,13 +35,33 @@ class FireDatabase : Database() {
     }
 
     override fun getUsernameFromUserId(userId: String): CompletableFuture<String> {
-        TODO("Will be implemented in the rebase database task")
+        val future = CompletableFuture<String>()
+
+        database.child(usersProfileFileName).child(userId).child("username").get().addOnSuccessListener {
+            if(it.value == null) future.completeExceptionally(NoSuchFieldException("There is no username corresponding to the userId $userId"))
+            else future.complete(it.value as String)
+        }.addOnFailureListener{
+            future.completeExceptionally(it)
+        }
+        return future
+    }
+
+    override fun getUserIdFromUsername(username: String): CompletableFuture<String> {
+        val future = CompletableFuture<String>()
+
+        database.child(usernameToUserIdFileName).child(username).get().addOnSuccessListener {
+            if(it.value == null) future.completeExceptionally(NoSuchFieldException("There is no userId corresponding to the username $username"))
+            else future.complete(it.value as String)
+        }.addOnFailureListener{
+            future.completeExceptionally(it)
+        }
+        return future
     }
 
     override fun isUsernameAvailable(userName: String): CompletableFuture<Boolean> {
         val future = CompletableFuture<Boolean>()
 
-        database.child("users").child(userName).get().addOnSuccessListener {
+        database.child(usernameToUserIdFileName).child(userName).get().addOnSuccessListener {
```

Comment:
`database.child(usernameToUserIdFileName)` is copied 4 times in this file. It should be refactored.